pkg/controller/s2irun/ksbuilder_jobs.go

```diff
@@ -278,37 +278,20 @@ func (r *ReconcileS2iRun) setGitSecret(instance *devopsv1alpha1.S2iRun, config *
 			return err
 		}
 
-		switch secret.Type {
-		case corev1.SecretTypeBasicAuth:
-			username, ok := secret.Data[corev1.BasicAuthUsernameKey]
-			if !ok {
-				return fmt.Errorf("could not get username in secret %s", secret.Name)
-			}
-			password, ok := secret.Data[corev1.BasicAuthPasswordKey]
-			if !ok {
-				return fmt.Errorf("could not get password in secret %s", secret.Name)
-			}
-			sourceUrl, err := url.Parse(config.SourceURL)
-			if err != nil {
-				return err
-			}
-			config.SourceURL = fmt.Sprintf("%s://%s:%s@%s%s", sourceUrl.Scheme, url.QueryEscape(string(username)), url.QueryEscape(string(password)), sourceUrl.Host, sourceUrl.RequestURI())
-
-		default:
-			username, ok := secret.Data[corev1.BasicAuthUsernameKey]
-			if !ok {
-				return fmt.Errorf("could not get username in secret %s", secret.Name)
-			}
-			password, ok := secret.Data[corev1.BasicAuthPasswordKey]
-			if !ok {
-				return fmt.Errorf("could not get password in secret %s", secret.Name)
-			}
-			sourceUrl, err := url.Parse(config.SourceURL)
-			if err != nil {
-				return err
-			}
-			config.SourceURL = fmt.Sprintf("%s://%s:%s@%s%s", sourceUrl.Scheme, url.QueryEscape(string(username)), url.QueryEscape(string(password)), sourceUrl.Host, sourceUrl.RequestURI())
+		username, ok := secret.Data[corev1.BasicAuthUsernameKey]
+		if !ok {
+			return fmt.Errorf("could not get username in secret %s", secret.Name)
+		}
+		password, ok := secret.Data[corev1.BasicAuthPasswordKey]
+		if !ok {
+			return fmt.Errorf("could not get password in secret %s", secret.Name)
+		}
+		sourceUrl, err := url.Parse(config.SourceURL)
+		if err != nil {
+			return err
 		}
+		config.SourceURL = fmt.Sprintf("%s://%s:%s@%s%s", sourceUrl.Scheme, url.QueryEscape(string(username)), url.QueryEscape(string(password)), sourceUrl.Host, sourceUrl.RequestURI())
```

Comment:
Should consider the case where no branch and commit.

Reply:
Here just formatted SourceURL，which does not have relationship with branch and commit. Branch and commit are set in S2iBuilder.spec，it  also works well even if they are not set.